packages/worker/src/workflow/workflow-worker-thread.ts

```diff
@@ -0,0 +1,86 @@
+import { isMainThread, parentPort as parentPortOrNull } from 'worker_threads';
+import { IllegalStateError } from '@temporalio/common';
+import { VMWorkflow, VMWorkflowCreator } from './vm';
+import { WorkerThreadRequest } from './workflow-worker-thread/input';
+import { WorkerThreadResponse } from './workflow-worker-thread/output';
+import { coresdk } from '@temporalio/proto';
+
+if (isMainThread) {
+  throw new IllegalStateError(`Imported ${__filename} from main thread`);
+}
+
+if (parentPortOrNull === null) {
+  throw new TypeError(`${__filename} got a null parentPort`);
+}
+
+// Create a new parentPort variable that is not nullable to please TS
+const parentPort = parentPortOrNull;
+
+function ok(requestId: BigInt): WorkerThreadResponse {
+  return { requestId, result: { type: 'ok' } };
+}
+
+let workflowCreator: VMWorkflowCreator | undefined;
+const workflowByRunId = new Map<string, VMWorkflow>();
+
+async function handleRequest({ requestId, input }: WorkerThreadRequest): Promise<WorkerThreadResponse> {
```

Comment:
This needs a docstring. The whole message passing convention and protocol should be given at least a cursory doc treatment

Reply:
(Though to be clear I'm not sure specifically this function is the best place, but probably the send and recv halves should refer to each other)

Reply:
Yes, we need way more docstrings in this whole threaded vm code.
I didn't invest too much in that initially because I wasn't sure if we'd use this code.
Forgot to add the docs before submitting the PR.
I'll do that now.